sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -186,26 +188,49 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
     }
   }
 
-  private async _createMessageSessionIfDoesntExist(): Promise<void> {
-    if (this._messageSession) {
-      return;
-    }
-    this._context.isSessionEnabled = true;
-    this._messageSession = await MessageSession.create(this._context, {
-      sessionId: this._sessionOptions.sessionId,
-      maxSessionAutoRenewLockDurationInSeconds: this._sessionOptions
-        .maxSessionAutoRenewLockDurationInSeconds,
-      receiveMode: convertToInternalReceiveMode(this.receiveMode)
-    });
-    // By this point, we should have a valid sessionId on the messageSession
-    // If not, the receiver cannot be used, so throw error.
-    if (this._messageSession.sessionId == null) {
-      const error = new Error("Something went wrong. Cannot lock a session.");
-      log.error(`[${this._context.namespace.connectionId}] %O`, error);
-      throw error;
+  private async _createMessageSessionIfDoesntExist(timeoutInMs: number): Promise<number> {
+    const createSessionOperationStartTime = Date.now();
+
+    const actionAfterTimeout = () => {
+      const desc: string = `Unable to create the message session in time. Please try again later.`;
+      const e: Error = {
+        name: "OperationTimeoutError",
+        message: desc
+      };
+
+      throw e;
+    };
+
+    const waitTimer = setTimeout(actionAfterTimeout, timeoutInMs);
```

Comment:
On timeout, we aren't really cancelling the creation of the `MessageSession`.
Event if we do end up throwing the error, the receiver link may end up being created behind the scenes.
I would say, for the sake of this PR, we remove the timeout for the create new session part, and re-visit it for Preview 2

Reply:
Hmmm.. We need to push the timeout to a lower level to throw the error only if the link is not created.

> And our second attempt might fail with SessionCannotBeLocked error, because the first attempt succeeded in getting a lock on the session.

I would like to see this in action.

Removed the timeout from `createMessageSession` but still kept it inside the OperationPromise.

Reply:
Isn't a similar argument applicable for the sender's initialization before sending the request ??
I'll look into the event-hubs version of `trySend`.